test/integration/notifications&auth/notifications-activation.test.tsx

```diff
@@ -0,0 +1,209 @@
+import { integrationTestRender } from '../../lib/render-helpers';
+import * as backgroundConnection from '../../../ui/store/background-connection';
+import {
+  act,
+  fireEvent,
+  waitFor,
+  screen,
+  within,
+} from '@testing-library/react';
+import { createMockImplementation } from '../helpers';
+import { getMockedNotificationsState } from './data/notification-state';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../shared/constants/metametrics';
+
+jest.mock('../../../ui/store/background-connection', () => ({
+  ...jest.requireActual('../../../ui/store/background-connection'),
+  submitRequestToBackground: jest.fn(),
+  callBackgroundMethod: jest.fn(),
+}));
+
+const backgroundConnectionMocked = {
+  onNotification: jest.fn(),
+};
+
+const mockedBackgroundConnection = jest.mocked(backgroundConnection);
+
+const setupSubmitRequestToBackgroundMocks = (
+  mockRequests?: Record<string, unknown>,
+) => {
+  mockedBackgroundConnection.submitRequestToBackground.mockImplementation(
+    createMockImplementation({
+      ...(mockRequests ?? {}),
+    }),
+  );
+};
+
+const trackNotificationsActivatedMetaMetricsEvent = async (
+  actionType: string,
+  profileSyncEnabled: boolean,
+) => {
+  await waitFor(() => {
+    const action =
+      mockedBackgroundConnection.submitRequestToBackground.mock.calls?.find(
+        (call) =>
+          call[0] === 'trackMetaMetricsEvent' &&
+          call[1]?.[0].category ===
+            MetaMetricsEventCategory.NotificationsActivationFlow &&
+          call[1]?.[0].properties.action_type === actionType,
+      );
+
+    expect(action?.[0]).toBe('trackMetaMetricsEvent');
+
+    const [event] = action?.[1] as unknown as [
+      {
+        event: string;
+        category: string;
+        properties: Record<string, unknown>;
+      },
+    ];
+
+    expect(event?.event).toBe(MetaMetricsEventName.NotificationsActivated);
+    expect(event?.category).toBe(
+      MetaMetricsEventCategory.NotificationsActivationFlow,
+    );
+    expect(event?.properties).toMatchObject({
+      action_type: actionType,
+      is_profile_syncing_enabled: profileSyncEnabled,
+    });
+  });
```

Comment:
I was doing some exploring, what are your thoughts on this?
```ts
const trackNotificationsActivatedMetaMetricsEvent = async (
  actionType: string,
  profileSyncEnabled: boolean,
) => {
  const expectedCall = [
    'trackMetaMetricsEvent',
    [
      expect.objectContaining({
        event: MetaMetricsEventName.NotificationsActivated,
        category: MetaMetricsEventCategory.NotificationsActivationFlow,
        properties: {
          action_type: actionType,
          is_profile_syncing_enabled: profileSyncEnabled,
        },
      }),
    ],
  ];

  expect(
    mockedBackgroundConnection.submitRequestToBackground.mock.calls,
  ).toStrictEqual(expect.arrayContaining([expectedCall]));
};
```

Reply:
If we adopt this, can we add it for the other assertions on background calls we make?

Reply:
Yes this is a good idea actually. I can adopt this. the object does have some other properties in this case though but this is what was in the requirements so I focused on them. Nevertheless, I like this approach

Reply:
Nicely done!